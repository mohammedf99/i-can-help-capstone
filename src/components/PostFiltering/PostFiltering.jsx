import React, { useEffect, useState } from "react";
import { useTranslation } from "next-i18next";
import { Row, Col } from "antd";
import { FilterContainer } from "./PostFiltering.styled";

const PostFiltering = ({ setFilteredPosts, posts, type }) => {
  const [filtering, setFiltering] = useState({
    employment: null,
    from: null,
    to: null,
    location: null,
    gender: null,
    type: null,
  });

  useEffect(() => {
    setFiltering((prev) => ({ ...prev, type }));
  }, [type]);

  const search = () => {
    setFilteredPosts(
      posts.filter(
        (post) =>
          (filtering.from ? Number(post.data().price) >= Number(filtering.from) : true) &&
          (filtering.to ? Number(post.data().price) <= (Number(filtering.to) || 1000000) : true) &&
          (filtering.employment ? post.data().employment === filtering.employment : true) &&
          (filtering.location ? post.data().location === filtering.location : true) &&
          (filtering.gender ? post.data().gender === filtering.gender : true) &&
          (filtering.type ? Number(post.data().type) === Number(filtering.type) : true),
      ),
    );
  };

  useEffect(() => search(), [filtering]);

  return (
    <FilterContainer>
      <form action="">
        <Row gutter={[8, 8]}>
          <Col span={24}>
            <h3 className="salary-title" style={{ backgroundColor: "#CACAEA", color: "#1C1259" }}>
              {t("salary")}
            </h3>
          </Col>
        </Row>
        <Row gutter={[8, 8]} className="salary-row">
          <Col span={4} className="from-to-title">
            <label htmlFor="from">{t("from")}</label>
          </Col>
          <Col span={8}>
            <input
              defaultValue={0}
              min={0}
              type="number"
              id="from"
              name="from"
              placeholder="$"
              className="salary-input"
              onChange={(e) => {
                e.persist();
                setFiltering((prev) => ({
                  ...prev,
                  from: e.target.value === "" ? null : e.target.value,
                }));
              }}
            />
          </Col>
          <Col span={4} className="from-to-title">
            <label htmlFor="to">{t("to")}</label>
          </Col>
          <Col span={8}>
            <input
              min={0}
              type="number"
              id="to"
              name="to"
              placeholder="$"
              className="salary-input"
              onChange={(e) => {
                e.persist();
                setFiltering((prev) => ({
                  ...prev,
                  to: e.target.value === "" ? null : e.target.value,
                }));
              }}
            />
          </Col>
        </Row>
        <Row gutter={[8, 8]} className="select-row">
          <Col span={24}>
            <select
              className="filter-select"
              onChange={(e) => {
                e.persist();
                setFiltering((prev) => ({
                  ...prev,
                  employment: e.target.value === "false" ? false : e.target.value,
                }));
              }}
            >
              <option selected disabled>
                {t("empType")}
              </option>
              <option value="Contract">Contract</option>
              <option value="Full-time">Full-time</option>
              <option value="Part-time">Part-time</option>
              <option value={false}>Show all</option>
            </select>
          </Col>
          <Col span={24}>
            <select
              className="filter-select"
              onChange={(e) => {
                e.persist();
                setFiltering((prev) => ({
                  ...prev,
                  location: e.target.value === "false" ? false : e.target.value,
                }));
              }}
            >
              <option selected disabled>
                {t("location")}
              </option>
              <option value="Erbil">Erbil</option>
              <option value="Duhok">Duhok</option>
              <option value="Halabja">Halabja</option>
              <option value="Kerkuk">Kerkuk</option>
              <option value="Sulaymaniyah">Sulaymaniyah</option>
              <option value={false}>Show all</option>
            </select>
          </Col>
          <Col span={24}>
            <select
              className="filter-select"
              onChange={(e) => {
                e.persist();
                setFiltering((prev) => ({
                  ...prev,
                  gender: e.target.value === "false" ? false : e.target.value,
                }));
              }}
            >
              <option selected disabled>
                {t("gender")}
              </option>
              <option value="Female">Female</option>
              <option value="Male">Male</option>
              <option value={false}>Show both</option>
            </select>
          </Col>
          <Col span={24}>
            <select
              className="filter-select"
              onChange={(e) => {
                e.persist();
                setFiltering((prev) => ({
                  ...prev,
                  type: e.target.value === "false" ? false : e.target.value,
                }));
              }}
            >
              <option selected disabled defaultValue={`${type}`}>
                {type === "1" ? "Employeer" : "Jobseeker"}
              </option>
              <option value="1">Employeer</option>
              <option value="2">Jobseeker</option>
              <option value={false}>Show both</option>
            </select>
          </Col>
        </Row>
      </form>
    </FilterContainer>
  );
};

export default PostFiltering;
