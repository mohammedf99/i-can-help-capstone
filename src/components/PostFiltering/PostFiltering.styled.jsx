import styled from "styled-components";

export const FilterContainer = styled.div`
  background-color: #fafaff;
  width: 100%;
  height: auto;
  padding: 1.25rem;
  border-radius: 5px;
  box-sizing: border-box;
  font-family: sans-serif;

  .salary-title {
    width: 100%;
    color: #4e4e9f;
    font-size: 1.25rem;
    font-weight: 700;
    padding: 0.25rem 0.5rem;
  }

  .salary-input {
    width: 100%;
    padding-left: 10px;
    font-size: 1rem;
    background-color: transparent;
    color: #1c1259;
  }

  input::-webkit-input-placeholder {
    color: #1c1259;
  }

  .salary-title,
  .salary-input {
    border: 2px solid #cacaea;
  }

  .salary-row,
  .salary-input {
    height: 40px;
  }

  .from-to-title {
    background-color: #cacaea;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1rem;
    font-weight: 600;
  }

  .salary-title,
  .salary-input,
  .from-to-title {
    border-radius: 5px;
  }

  .select-row {
    margin-top: 0.5rem;
  }

  .filter-select {
    width: 100%;
    height: 40px;
    border: 2px solid #cacaea;
    border-radius: 5px;
    background-color: transparent;
    color: #4e4e9f;
    font-size: 1rem;
    font-weight: 600;
  }
`;
