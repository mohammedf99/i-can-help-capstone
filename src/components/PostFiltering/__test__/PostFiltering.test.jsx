import * as React from "react";
import * as renderer from "react-test-renderer";
import PostFiltering from "../PostFiltering";

it("renders Post filtering correctly", () => {
  const tree = renderer.create(<PostFiltering />).toJSON();
  expect(tree).toMatchSnapshot();
});