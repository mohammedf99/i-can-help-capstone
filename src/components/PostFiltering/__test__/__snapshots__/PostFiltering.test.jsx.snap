// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Post filtering correctly 1`] = `
<div
  className="PostFilteringstyled__FilterContainer-sc-1kfb96c-0 gbZFuv"
>
  <form
    action=""
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginBottom": -4,
          "marginLeft": -4,
          "marginRight": -4,
          "marginTop": -4,
        }
      }
    >
      <div
        className="ant-col ant-col-24"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <h3
          className="salary-title"
          style={
            Object {
              "backgroundColor": "#CACAEA",
              "color": "#1C1259",
            }
          }
        >
          salary
        </h3>
      </div>
    </div>
    <div
      className="ant-row salary-row"
      style={
        Object {
          "marginBottom": -4,
          "marginLeft": -4,
          "marginRight": -4,
          "marginTop": -4,
        }
      }
    >
      <div
        className="ant-col ant-col-4 from-to-title"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <label
          htmlFor="from"
        >
          from
        </label>
      </div>
      <div
        className="ant-col ant-col-8"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <input
          className="salary-input"
          defaultValue={0}
          id="from"
          min={0}
          name="from"
          onChange={[Function]}
          placeholder="$"
          type="number"
        />
      </div>
      <div
        className="ant-col ant-col-4 from-to-title"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <label
          htmlFor="to"
        >
          to
        </label>
      </div>
      <div
        className="ant-col ant-col-8"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <input
          className="salary-input"
          id="to"
          min={0}
          name="to"
          onChange={[Function]}
          placeholder="$"
          type="number"
        />
      </div>
    </div>
    <div
      className="ant-row select-row"
      style={
        Object {
          "marginBottom": -4,
          "marginLeft": -4,
          "marginRight": -4,
          "marginTop": -4,
        }
      }
    >
      <div
        className="ant-col ant-col-24"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <select
          className="filter-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            empType
          </option>
          <option
            value="Contract"
          >
            Contract
          </option>
          <option
            value="Full-time"
          >
            Full-time
          </option>
          <option
            value="Part-time"
          >
            Part-time
          </option>
          <option
            value={false}
          >
            Show all
          </option>
        </select>
      </div>
      <div
        className="ant-col ant-col-24"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <select
          className="filter-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            location
          </option>
          <option
            value="Erbil"
          >
            Erbil
          </option>
          <option
            value="Duhok"
          >
            Duhok
          </option>
          <option
            value="Halabja"
          >
            Halabja
          </option>
          <option
            value="Kerkuk"
          >
            Kerkuk
          </option>
          <option
            value="Sulaymaniyah"
          >
            Sulaymaniyah
          </option>
          <option
            value={false}
          >
            Show all
          </option>
        </select>
      </div>
      <div
        className="ant-col ant-col-24"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <select
          className="filter-select"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            gender
          </option>
          <option
            value="Female"
          >
            Female
          </option>
          <option
            value="Male"
          >
            Male
          </option>
          <option
            value={false}
          >
            Show both
          </option>
        </select>
      </div>
      <div
        className="ant-col ant-col-24"
        style={
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <select
          className="filter-select"
          onChange={[Function]}
        >
          <option
            defaultValue="undefined"
            disabled={true}
            selected={true}
          >
            Jobseeker
          </option>
          <option
            value="1"
          >
            Employeer
          </option>
          <option
            value="2"
          >
            Jobseeker
          </option>
          <option
            value={false}
          >
            Show both
          </option>
        </select>
      </div>
    </div>
  </form>
</div>
`;
